WeakAuras/WeakAuras.lua

```diff
@@ -3185,51 +3188,60 @@ function Private.HandleGlowAction(actions, region)
     or (actions.glow_frame_type == "FRAMESELECTOR" and actions.glow_frame)
   )
   then
-    local glow_frame
+    local glow_frame, should_glow_frame
     if actions.glow_frame_type == "FRAMESELECTOR" then
       if actions.glow_frame:sub(1, 10) == "WeakAuras:" then
         local frame_name = actions.glow_frame:sub(11)
         if regions[frame_name] then
           glow_frame = regions[frame_name].region
+          should_glow_frame = true
         end
       else
         glow_frame = Private.GetSanitizedGlobal(actions.glow_frame)
+        should_glow_frame = true
       end
     elseif actions.glow_frame_type == "UNITFRAME" and region.state.unit then
       glow_frame = WeakAuras.GetUnitFrame(region.state.unit)
+      should_glow_frame = true
     elseif actions.glow_frame_type == "NAMEPLATE" and region.state.unit then
       glow_frame = WeakAuras.GetUnitNameplate(region.state.unit)
+      should_glow_frame = true
     end
 
-    if glow_frame then
+    if should_glow_frame then
       local id = region.id .. (region.cloneId or "")
       if actions.glow_action == "show" then
         -- remove previous glow
-        if region.active_glows_hidefunc
-        and region.active_glows_hidefunc[glow_frame]
-        then
-          region.active_glows_hidefunc[glow_frame]()
+        if glow_frame then
+          if region.active_glows_hidefunc
+          and region.active_glows_hidefunc[glow_frame]
+          then
+            region.active_glows_hidefunc[glow_frame]()
+          end
```

Comment:
I have not looked to deep into it, but should the hiding of the old frame glow depend on figuring out the new frame?
Or should this code be moved outside the if glow_frame check?

I see that it was previously guarded by the same check, so keeping it that way is probably fine. But it looks a bit odd to me,

Reply:
Problem of running this hide glow function outside of the `if glow_frame` check is that a region can apply a glow to multiple frames at once, so which one should be stop?

This code block is only to avoid having 2 different glow applied on the same frame by same region, hide glow func only support one glow per frame/region